llvm/lib/CodeGen/SelectionDAG/SelectionDAGISel.cpp

```diff
@@ -142,6 +142,12 @@ UseMBPI("use-mbpi",
         cl::init(true), cl::Hidden);
 
 #ifndef NDEBUG
+static cl::list<std::string> FilterDAGFuncNames(
+    "filter-print-dags-funcs", cl::Hidden, cl::value_desc("function names"),
+    cl::desc("Only print DAGs of functions whose names "
```

Comment:
I think we should take a step back and figure out what the goal of `-debug-only=isel -filter-print-dags-funcs=` is.

This patch seems to have touched every debug message in this file, but this isn't the only SelectionDAG file.

Are you just wanting to see the dumps of the DAG from  `CodeGenAndEmitDAG()` for a specific function? Do you care about any of the other messages for that function? Its not clear that you'd only want those debug messages in this file and not the other SelectionDAG files. Since the split between files isn't completely clean.

If you only want to see the dumps from `CodeGenAndEmitDAG()`, then maybe those should be given their own DEBUG_TYPE separate from the rest of this file. So you could do `-debug-only=isel-dumps -filter-print-dags-funcs=`

If you want all debug messages for a particular function then maybe this property should live on SelectionDAG and filter all of the messages in all of the SelectionDAG related files?

Reply:
Seems like this should have a scope very similar to `FilterDAGBasicBlockName`. I'm also a bit confused why we need an new option, since it feels very similar to `-filter-print-funcs=`... can we just reuse that? IMO asking for a dump of the DAG + `-filter-print-funcs=` seems natural, but maybe it's missing the desired granularity, or would it include information you'd like to exclude from the output?

Reply:
> If you only want to see the dumps from CodeGenAndEmitDAG(), then maybe those should be given their own DEBUG_TYPE separate from the rest of this file. So you could do -debug-only=isel-dumps -filter-print-dags-funcs=

Printing traces in CodeGenAndEmitDAG is my primary goal, so I'm probably going with the `-debug-only=isel-dumps` + filter function flag approach.

> it feels very similar to -filter-print-funcs=... can we just reuse that? IMO asking for a dump of the DAG + -filter-print-funcs= seems natural

Sounds like a good idea to me. I just found that several CodeGen Passes are also using `-filter-print-funcs`.

> but maybe it's missing the desired granularity, or would it include information you'd like to exclude from the output?

I don't think these will be a problem.